datatree/treenode.py

```diff
@@ -601,15 +601,18 @@ def name(self, name: str | None) -> None:
                 raise TypeError("node name must be a string or None")
             if "/" in name:
                 raise ValueError("node names cannot contain forward slashes")
+        parent = self._parent
+        self.orphan()
         self._name = name
+        self._set_parent(parent, name)
```

Comment:
Instead of this orphaning then un-orphaning logic, couldn't we just change the key in `.children` directly via some logic more like in `_post_attach`

Reply:
This does seem immediate to me right now (I tried some things but not successfully until now)
I still need to remove then add again the node in the parent's children at some point :thinking: